ADR/ADR037-implement-review-apps.md

```diff
@@ -0,0 +1,200 @@
+
+### Why do we want them
+
+We want to introduce review apps on PRs to support both technical and non-technical team members in gaining access to a working version of the code in question. For non-technical team members, introducing review apps will make it possible for them to make changes and test changes (usually content changes) for themselves, without the support of a developer. For engineer team members, review apps will make it easier for them to test a subset of the system in a production-ish environment without needing to disturb the fully fledged development environment.
+
+### What is in use elsewhere in GDS?
+
+Other teams in GDS have faced the same issues as us. One particularly good example is the GOV.UK Design System team, who have a mix of technical and non-technical members. They host prototypes and review versions of applications on [Heroku](https://www.heroku.com/) using Heroku's [Review Apps](https://devcenter.heroku.com/articles/github-integration-review-apps) feature. This is also used on GOV.UK Forms among our content and interaction designers for hosting prototypes.
+
+We investigated using Heroku Review Apps for our purposes, but we concluded that it was not a good choice. Heroku Review Apps is aimed at use cases involving a single application, but the applications we want to get review apps for will require deploying more than one application at a time.
+
+## Decision
+
+In this section we describe the architectural decisions we've made. Implementation details are out of scope.
+
+### Workflow 
+
+We have decided that the workflow for working with review apps should look like this
+
+```mermaid
+%%{init: { "sequence": { "showSequenceNumbers": true }}}%%
+        
+sequenceDiagram
+
+actor dev as Dev
+actor reviewer as Reviewer
+participant pr as PR
+participant app as Review App
+
+dev ->> pr: open
+activate pr
+
+pr ->> app: create
+activate app
+
+pr -->> dev: review app URL
+
+loop Iteration
+    dev ->> pr: new commit
+    pr ->> app: redeploy
+    dev ->> app: visit
+end
+
+reviewer ->> pr: review
+pr -->> reviewer: review app URL
+reviewer ->> app: visit
+reviewer ->> pr: approve
+
+dev ->> pr: merge
+pr ->> app: destroy
+deactivate app
+
+pr -->> dev: merged
+deactivate pr
+```
+
+A developer (or non-technical team member) will raise a pull request for their change, and the system we build will deploy a review app in response to the opening of the pull request. The system will share a link to the running app in a comment on the pull request.
+
+Subsequent changes to the pull request will result in the app being redeployed.
+
+Once a reviewer has reviewed and merged the pull request, the system will destroy the running applications. 
+
+### Architecture 
```

Comment:
Again, these diagrams are useful but probably should live outside the ADR so they can be evolved without having to update the ADR.